x/twap/logic_test.go

```diff
@@ -85,6 +85,66 @@ func (s *TestSuite) TestGetSpotPrices() {
 	}
 }
 
+func (s *TestSuite) TestIntegrationForTwap() {
+	poolAssets := sdk.NewCoins(sdk.NewInt64Coin(denom0, 5), sdk.NewInt64Coin(denom1, 25))
+	poolId := s.PrepareBalancerPoolWithCoins(poolAssets...)
+
+	s.twapkeeper.EndBlock(s.Ctx)
+	s.Commit()
+
+	// we're going to use this for twap querying later on
+	oldTime := s.Ctx.BlockTime()
+	twapRecord, err := s.twapkeeper.GetRecordAtOrBeforeTime(s.Ctx, poolId, s.Ctx.BlockTime(), denom0, denom1)
+	s.Require().NoError(err)
+
+	s.Require().Equal(twapRecord.P0LastSpotPrice, sdk.MustNewDecFromStr("5"))
```

Comment:
instead of hardcoding the values "5" and "0.2" throughout the test, we might want to define reusable constants to make it clearer where the values are coming from

Reply:
changed it to variables where we derive the value by dividing the amount of pool assets!